kmerexpr/fasta.py

```diff
@@ -0,0 +1,43 @@
+"""
+Module for reading in FASTA sequence files.
+
+Structured similarly to BioPython's BSD-3 parser
+https://github.com/biopython/biopython/blob/76796c970a68fa09da94dc95827b217a872004e2/Bio/SeqIO/FastaIO.py
+"""
+
+from dataclasses import dataclass
+from typing import Generator
+from io import TextIOBase
+
+
+@dataclass
+class Sequence:
+    header: str
+    sequence: str
+
+
+def read_fasta(handle: TextIOBase) -> Generator[Sequence, None, None]:
+    """
+    Basic lazy FASTA format reader.
+
+    This assumes the only comments are headers. Sequences can be split
```

Comment:
I'm not sure what "only comments are headers" means.  Looking at the code, lines before the first line starting with `>` are just skipped and that after that, everything is either part of a header or part of a sequence.  Is that what this means?  If that's right, can we say that more directly?

Reply:
[Wikipedia](https://en.wikipedia.org/wiki/FASTA_format) has some examples of sequences with multiple comments, something like 

```
> Some header
> Some second comment
GATTACA
```
The idea is that the first line is the header, the second line is ignored entirely, and the third line is the sequence. 

This is valid under "the original Pearson FASTA format", but not "the NCBI FASTA specfication", according to the article. 

As far as I can tell, none of the open source parsers will accept it.